spec/app_spec.rb

```diff
@@ -0,0 +1,117 @@
+    expect(terminal).to receive(:puts).with("Here's a list of all orders:").ordered
+    expect(terminal).to receive(:puts).with("").ordered
+    expect(order_repository).to receive(:all).and_return [order1, order2, order3]
+    expect(order_repository).to receive(:find_with_item).with(1).and_return order1
+    expect(terminal).to receive(:puts).with("#1 Rob - Date: 2022-01-01").ordered
+    expect(terminal).to receive(:puts).with("   TV").ordered
+    expect(terminal).to receive(:puts).with("   Fridge").ordered
+    expect(order_repository).to receive(:find_with_item).with(2).and_return order2
+    expect(terminal).to receive(:puts).with("#2 Tom - Date: 2022-01-02").ordered
+    expect(terminal).to receive(:puts).with("   TV").ordered
+    expect(order_repository).to receive(:find_with_item).with(3).and_return order3
+    expect(terminal).to receive(:puts).with("#3 Anisha - Date: 2022-01-02").ordered
+    expect(terminal).to receive(:puts).with("   Toaster").ordered
+    app = Application.new("shop_inventory_test", terminal, item_repository, order_repository)
+    app.run
+  end
+
+  it "creates an item if user input is 2" do
+    terminal = double :terminal
+    item4 = double :item, id: 4, name: "Kettle", price: 8.99, quantity: "13"
+    item_repository = double :item_repository, create: item4
+    order_repository = double :order_repository
+    expect(terminal).to receive(:puts).with("Welcome to the shop management program!").ordered
+    expect(terminal).to receive(:puts).with("").ordered
+    expect(terminal).to receive(:puts).with("What do you want to do?").ordered
+    expect(terminal).to receive(:puts).with("  1 = list all shop items").ordered
+    expect(terminal).to receive(:puts).with("  2 = create a new item").ordered
+    expect(terminal).to receive(:puts).with("  3 = list all orders").ordered
+    expect(terminal).to receive(:puts).with("  4 = create a new shop order").ordered
+    expect(terminal).to receive(:puts).with("").ordered
+    expect(terminal).to receive(:gets).and_return("2").ordered
+    expect(terminal).to receive(:puts).with("").ordered
+    expect(terminal).to receive(:puts).with("Item name:").ordered
+    expect(terminal).to receive(:gets).and_return("Kettle").ordered
+    expect(terminal).to receive(:puts).with("Item price:").ordered
+    expect(terminal).to receive(:gets).and_return(8.99).ordered
+    expect(terminal).to receive(:puts).with("Item quantity:").ordered
+    expect(terminal).to receive(:gets).and_return(13).ordered
+    app = Application.new("shop_inventory_test", terminal, item_repository, order_repository)
+    app.run
+  end
+
+  it "creates an order if user input is 4" do
+    terminal = double :terminal
+    item1 = double :item, id: 1, name: "TV", price: 99.99, quantity: "5"
+    item_repository = double :item_repository, find: item1
+    order4 = double :order, id: 4, customer: 'Shah', date: '2022-10-10'
+    order_repository = double :order_repository, create: order4
```

Comment:
Great use of doubles to test that the application was appropriately using the create methods.